### PR TITLE
Add back NuGet.Packaging.Core with TypeForwardedTo attrs

### PR DESCRIPTION
**Pull request by rrelyea**
## Bug

Fixes: https://github.com/NuGet/Home/issues/7768 
Regression: Yes

## Fix

Details: 
We recently removed NuGet.Packaging.Core.dll and moved all of the code into NuGet.Packaging.dll

This is causing problems for old tools that run in msbuild/dotnetCLI.

We'll add the assembly back in, but keep all the code move.
We'll add typeforwardedto attrs to point to the new location.
(since that is done with typeof(*), we need N.P.C to depend on N.P.

## Testing/Validation

Tests Added: Yes/No  
Reason for not adding tests:  
Validation:  

TODO: 
does it build nupkg in full build?
- [X] It does. The nupkg is created with correct metadata.

how will CLI need to adapt? (take a top level dependency on this DLL?)
- [ ] They will need to take a top level dependency wherever they face this problem. 

will we need to install into VSIX?
- [ ] It's not clear to me whether we need it. 
We have inserted into VS and have not heard any feedback yet. Don't think we can be certain whether anyone is referencing NuGet.packaging.Core directly.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging.Core/NuGet.Packaging.Core.csproj [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
+  <PropertyGroup>
+    <Description>The (former home to) core data structures for NuGet.Packaging.</Description>
+    <TargetFrameworks>$(TargetFrameworksLibrary)</TargetFrameworks>
+    <TargetFramework />
+    <NoWarn>$(NoWarn);CS1591</NoWarn>
+    <PackProject>true</PackProject>
+    <Shipping>true</Shipping>
+    <IncludeInVSIX>true</IncludeInVSIX>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.Packaging.Core/NuGet.Packaging.Core.csproj]: This should be removed. 
It ends up including the localization assets from the nuget.packaging.core.dll in VSIX. 

The nuget.packaging.core.dll is not there in the VSIX (and as far as I know it doens't need to be)

**Review by nkolev92 — COMMENTED** [review]
I have pushed some changes to cleanup the VSIX build. 

I will double check all the assets when the build completes.

**Comment by nkolev92** [comment]
Everything looks fine.
Both VSIXs are built fine, no extra files in there. 
The package is created and signed correctly.

**Review by nkolev92 — APPROVED** [review]

**Comment by rrelyea** [comment]
👍 
yes, i agree, let's keep it out of VSIX for now.
Let's discuss at standup. Likely we put in dev branch, and consider Livar's latest email. If CLI cannot take this in p3, do we shoot for p4 with it?

**Pull request merged**